File: Angular7-WebApp/src/Models/Employee.ts

export class Employee{
    firstname:string;
    lastname:string ;
    email:string;
    gender:number;
    id:string;

    constructor(){};

    // constructor (firstname: string, lastname: string, email: string, gender:number, id:string) {
    //     this.id = id;
    //     this.firstname = firstname;
    //     this.lastname = lastname;
    //     this.email = email;
    //     this.gender = gender;
    // }
}

export class EmployeeResponse {
     employee: any;
     fileToUpload: FileToUpload;
     constructor(emp: any, filesData: FileToUpload){
        this.employee = emp;
        this.fileToUpload = filesData;
     };
}

export class FileToUpload {
    fileName: string = "";
    fileSize: number = 0;
    fileType: string = "";
    lastModifiedTime: number = 0;
    lastModifiedDate: Date = new Date();
    fileAsBase64: string = "";
  }

File: Angular7-WebApi-RestServices/Angular6DemoServices/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;


namespace Angular7DemoServices
{
    public class Employee
    {
        [Key]
        public int ID { get; set; }
        public string Fname { get; set; }
        public string Lname { get; set; }
        public string email { get; set; }
        public string gender { get; set; }
        public int? IDDocument { get; set; }

        public Employee() { }

        public Employee(int EmpId, string EmpName, string EmpLast, string EmailId, string Gender)
        {
            this.ID = EmpId;
            this.Fname = EmpName;
            this.Lname = EmpLast;
            this.email = EmailId;
            this.gender = Gender;
        }
    }

    public class file
    {
        public int Id { get; set; }
        [Required]
        [StringLength(255)]
        public string FileName { get; set; }
    }

    public class EmployeeResponse {
        public Employee employee { get; set; }
        public FileToUpload fileToUpload { get; set; }
    }

    public class FileToUpload
    {
        [Key]
        public int ID { get; set; }
        public string FileName { get; set; }
        public string FileSize { get; set; }
        public string FileType { get; set; }
        public long LastModifiedTime { get; set; }
        public DateTime LastModifiedDate { get; set; }
        public string FileAsBase64 { get; set; }
        public byte[] FileAsByteArray { get; set; }
        public int IDEmployee { get; set; }

    }
}
